bundles/org.openhab.binding.tacmi/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,172 @@
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="coe-bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>TA C.M.I. CoE Connection</label>
+		<description>CoE Communication to the "Technische Alternative C.M.I. Control and Monitoring Interface"</description>
+
+		<config-description>
+			<parameter name="host" type="text" required="true">
+				<label>C.M.I. IP Address</label>
+				<description>Host name of IP address of the CMI</description>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="node" type="integer" min="1" max="62" required="true">
+				<label>Node</label>
+				<description>The CoE / CAN Node number openHAB should represent</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<!-- CoE digital Channel Type -->
+	<channel-type id="coe-digital-in">
+		<item-type>Switch</item-type>
+		<label>Digital Input (C.M.I. -> OH)</label>
+		<description>A digital channel sent from C.M.I. to openHAB</description>
+		<state readOnly="true"/>
+		<config-description>
+			<parameter name="output" type="integer" min="1" max="32" required="true">
+				<label>Output</label>
+				<description>C.M.I. Network Output</description>
+			</parameter>
+		</config-description>
+	</channel-type>
+	<channel-type id="coe-digital-out">
+		<item-type>Switch</item-type>
+		<label>Digital Output (OH -> C.M.I.)</label>
+		<description>A digital channel sent from OpenHAB to C.M.I.</description>
+		<config-description>
+			<parameter name="output" type="integer" min="1" max="32" required="true">
+				<label>Output</label>
+				<description>Network Output</description>
+			</parameter>
+			<parameter name="initialValue" type="boolean" required="false">
+				<label>Initial Value</label>
+				<description>Initial value to set after startup (optional, defaults to uninitialized)</description>
+			</parameter>
+		</config-description>
+	</channel-type>
+	<!-- COA analog Channel Type -->
+	<channel-type id="coe-analog-in">
+		<item-type>Number</item-type>
+		<label>Analog Input Channel (C.M.I. -> OH)</label>
+		<description>A Analog Channel received from the C.M.I.</description>
+		<state readOnly="true"/>
+		<config-description>
+			<parameter name="output" type="integer" min="1" max="32" required="true">
+				<label>Output</label>
+				<description>C.M.I. Network Output</description>
+			</parameter>
+		</config-description>
+	</channel-type>
+	<channel-type id="coe-analog-out">
+		<item-type>Number</item-type>
+		<label>Analog Output Channel (OH -> C.M.I.)</label>
+		<description>A Analog Channel sent to the C.M.I.</description>
+		<config-description>
+			<parameter name="output" type="integer" min="1" max="32" required="true">
+				<label>Output</label>
+				<description>Network Output</description>
+			</parameter>
+			<parameter name="type" type="integer" min="0" max="21" required="true">
+				<label>Measurement Type</label>
+				<description>Measurement type for this channel</description>
+				<options>
```

Comment:
Did you consider creating type specific channels for measurement type? Like a `Number:Temperature` channel for value 1 type channels. That way the user can use it as an actual temperature channel.

Reply:
Not yet - but yes, it's quite obvious and sounds like a good idea. Would it be okay to do this in a later PR? I also plan to find out all "UNKNOWN" types from [TACmiMeasureType.java](https://github.com/openhab/openhab-addons/pull/7768/files#diff-5b4052a0a557d600bc4580acabe1c010) to map to the real units. During this I also could create separate channels for all these different types.